### PR TITLE
Avoid writing metadata with default value.

### PR DESCRIPTION
**Pull request by niklasharrysson**

### CODE REVIEW

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 1]
```diff
@@ -146,6 +152,14 @@ namespace
             std::string valueString = md->getValueString();
             if (!valueString.empty())
             {
+                // Check if the value is the default value and the metadata could be ignored.
+                auto it = defaultValuedMetadata.find(name);
+                if (it != defaultValuedMetadata.end() && it->second == valueString)
```
Comment (niklasharrysson) [inline source/MaterialXRuntime/RtFileIo.cpp:157]: This makes it so we avoid writing out uivisible=true on all inputs. When the attribute is not set we assume the input is visible.

Comment (feldstj) [reply source/MaterialXRuntime/RtFileIo.cpp:157]: But what if a nodedef has a value of false for uivisible and the instance has a value of true? We still need to write out uivisible="true" in that case, no? Possible I'm missing something.

**Comment by bernardkwok** [comment]
Just curious about this one. Has it been resolved? 
I still get "uivisible=true" showing up all the time :).

**Comment by niklasharrysson** [comment]
Yeah this PR got stuck a while back.. I'll see if I can revive it :)

File: source/MaterialXRuntime/RtFileIo.cpp (hunk 1 above)
Comment (niklasharrysson) [reply source/MaterialXRuntime/RtFileIo.cpp:157]: The writeNode function has existing logic to write out any input where the uivisible differ from the corresponding nodedef input. So such inputs are always written out.

So if _"no uivisible found"_ is equal to uivisible="true", then we are good. So in the case the nodedef input say uivisible="false" the existence of this node input, and the missing uivisible, means that it is overriding the nodedef input by being visible.

**Review by feldstj — APPROVED** [review]

**Comment by niklasharrysson** [comment]
Closing this as I think we should take a more holistic look at how to avoid writing metadata with default value.

**Pull request closed without merging**